VersionControlUI/src/DiffManager.cpp

```diff
@@ -182,14 +208,119 @@ void DiffManager::visualize()
+		auto newCommit = diffSetup.repository_->getCommitInformation(diffSetup.newVersion_);
+		QString message = createHTMLCommitInfo(newCommit);
+		auto overlay = new Visualization::MessageOverlay{diffViewItem,
+				[diffViewItem, message](Visualization::MessageOverlay* overlay)
+		{
+			auto diffViewItemPos = diffViewItem->scenePos();
+			overlay->setPos(diffViewItemPos.x(),
+								 diffViewItemPos.y() - overlay->heightInScene());
+			auto vDiffComparisonPair = DCast<VersionControlUI::VDiffComparisonPair>(diffViewItem->findVisualizationOf
+																					(diffViewItem->nodeAt(Visualization::MajorMinorIndex{})));
+			overlay->setScale(vDiffComparisonPair->scaleFactor());
+			return message;
+		}};
+
+
+		diffViewItem->addOverlay(overlay, "DiffInfoMessageOverlay");
 	});
 
 	// switch to the newly created view
 	Visualization::VisualizationManager::instance().mainScene()->viewItems()->switchToView(diffViewItem);
 }
 
+QString DiffManager::createHTMLCommitInfo(FilePersistence::CommitMetaData commitMetaData)
+{
+	return commitMetaData.message_ + "<br/><br/>"
+			+ "<font color='gray'>" + commitMetaData.author_.name_ + "</font><br/>"
+			+ "<font color='gray'>" + commitMetaData.dateTime_.toString("dd.MM.yyyy hh:mm") + "</font><br/>"
+			+ "<font color='gray'>" + commitMetaData.sha1_ + "</font>";;
+}
+
+void DiffManager::visualizeHistory(Model::NodeIdType targetNodeID, QList<QString> versions)
+{
+
+	targetNodeID_ = targetNodeID;
+
+	auto historyViewItem = Visualization::VisualizationManager::instance().mainScene()->
+			viewItems()->viewItem("HistoryView");
+
+	if (historyViewItem)
+		Visualization::VisualizationManager::instance().mainScene()->
+					viewItems()->removeViewItem(historyViewItem);
+
+	historyViewItem = Visualization::VisualizationManager::instance().mainScene()->
+				viewItems()->newViewItem("HistoryView");
+
+	historyViewItem->setMajorAxis(Visualization::GridLayouter::NoMajor);
+	historyViewItem->setZoomLabelsEnabled(false);
+
+	int col = 0;
+
+	QList<QPair<Model::Node*, QString>> diffComparisonPairInfo;
+
+	for (int i = 1; i < versions.length(); i++)
+	{
+		DiffSetup diffSetup;
+
+		// detailed changes
+		QList<ChangeWithNodes> changesWithNodes;
+
+		// contains the nodes which will be drawn
+		QSet<Model::NodeIdType> changedNodesToVisualize;
+
+		QString oldVersion = versions[i-1];
+		QString newVersion = versions[i];
+
+		// fill up lists
+		computeDiff(oldVersion, newVersion, changesWithNodes, changedNodesToVisualize, diffSetup);
+
+		Visualization::VisualizationManager::instance().mainScene()->listenToTreeManager(diffSetup.newVersionManager_);
+		Visualization::VisualizationManager::instance().mainScene()->listenToTreeManager(diffSetup.oldVersionManager_);
+
+		QString message = createHTMLCommitInfo(diffSetup.repository_->getCommitInformation(diffSetup.newVersion_));
+
+		int row = 0;
+		for (auto diffComparisonPair : createDiffComparisonPairs(diffSetup, changedNodesToVisualize))
```

Comment:
I'm a bit confused. Isn't it the case that for history there will always be exactly one pair? Why is there a loop here?